src/rgw/rgw_group.h

```diff
@@ -121,8 +121,8 @@ class RGWGroup
   int update();
   int update_name(const string& new_name);
   int update_path(const string& new_path);
-  bool add_user(string& user);
-  bool remove_user(string& user);
+  bool add_user(const string& user);
+  bool remove_user(const string& user);
```

Comment:
@adamemerson : string_view wont help here, since no parsing is being done here, strings are just being added to a vector. Let me know if you think otherwise.

Reply:
@adamemerson : The ARN class doesn't take into account the 'path' of groups/ roles/ users. Can I make changes to the class to incorporate the path also? Then it may not be needed to store the ARN alongwith other info.

Reply:
Oh, yes. Feel free to change the ARN class or anything else as you see fit.

(Is it the ARN class that's missing the capability or the Identity class? ARN /should/ be able to support any ARN and if it can't I would definitely appreciate it if you make sure it can or point out what it's missing to me, whichever is more convenient for you. The Identity class I know doesn't have a notion of paths for users/groups/etc..)